src/coreclr/nativeaot/Runtime/amd64/GC.asm

```diff
@@ -19,3 +19,20 @@ LEAF_ENTRY xmmYmmStateSupport, _TEXT
 LEAF_END xmmYmmStateSupport, _TEXT
 
         end
+
+;; extern "C" DWORD __stdcall zmmStateSupport();
+LEAF_ENTRY zmmStateSupport, _TEXT
```

Comment:
Should this be named `xmmYmmZmmStateSupport` since its checking all 3-bits?

Reply:
Maybe, alternatively, `avx512StateSupport` is a better name since this this is actually checking `Opmask state`, `ZMM_Hi256 state`, and `Hi16_ZMM state`.

Reply:
Changed to `avx512StateSupport`